leftwm-core/src/handlers/command_handler/scratchpad_handler.rs

```diff
@@ -0,0 +1,1242 @@
+    let window_handle = window.or(manager
+        .state
+        .focus_manager
+        .window_history
+        .get(0)?
+        .as_ref()
+        .copied())?;
+
+    // retrieve and prepare window information
+    let window_pid = {
+        let ws = manager
+            .state
+            .focus_manager
+            .workspace(&manager.state.workspaces)?;
+        let to_scratchpad = manager
+            .state
+            .scratchpads
+            .iter()
+            .find(|s| s.name == scratchpad)?;
+        let new_float_exact = scratchpad_xyhw(&ws.xyhw, to_scratchpad);
+
+        let window = manager
+            .state
+            .windows
+            .iter_mut()
+            .find(|w| w.handle == window_handle)?;
+
+        // Put window in correct position
+        window.set_floating(true);
+        window.normal = ws.xyhw;
+        window.set_floating_exact(new_float_exact);
+        log::debug!("Set window to floating: {:?}", window);
+
+        window.pid?
+    };
+
+    if let Some(windows) = manager.state.active_scratchpads.get_mut(&scratchpad) {
+        log::debug!("Scratchpad {} already active, push scratchpad", &scratchpad);
+        let previous_scratchpad_handle = manager
+            .state
+            .windows
+            .iter()
+            .find(|w| w.pid.as_ref() == windows.front())
+            .map(|w| w.handle);
+
+        // check if window already in scratchpad
+        if windows.iter().any(|pid| *pid == window_pid) {
+            return Some(false);
+        }
+
+        windows.push_front(window_pid);
+        if let Some(previous_scratchpad_handle) = previous_scratchpad_handle {
+            hide_scratchpad(manager, &previous_scratchpad_handle).ok()?; // first hide current scratchpad window
+        }
+    } else {
+        log::debug!("Scratchpad {} not active yet, open scratchpad", &scratchpad);
+        manager
+            .state
+            .active_scratchpads
+            .insert(scratchpad, VecDeque::from([window_pid]));
+    }
+    manager.state.sort_windows();
+
+    Some(true)
+}
+
+/// Release a scratchpad to become a normal window. When tag is None, use current active tag as the
+/// destination. Window can be a handle to select a specific window, the name of a scratchpad or
+/// none to select the current window.
+pub fn release_scratchpad<C: Config, SERVER: DisplayServer>(
+    window: ReleaseScratchPadOption,
+    tag: Option<TagId>,
+    manager: &mut Manager<C, SERVER>,
+) -> Option<bool> {
+    let destination_tag =
+        tag.or_else(|| manager.state.focus_manager.tag_history.get(0).copied())?;
+
+    // if None, replace with current window
+    let window = if window == ReleaseScratchPadOption::None {
+        ReleaseScratchPadOption::Handle(
+            manager
+                .state
+                .focus_manager
+                .window_history
+                .get(0)?
+                .as_ref()
+                .copied()?,
+        )
```

Comment:
I think it would look nicer if this is assigned to a variable first

Reply:
This would be not possible as clippy would give the warning `clippy::let_and_return`

Reply:
Hm... I'd say clippy to allow this in this block because it does make it more readable if this would be in a variable or what do you think?